fe/lib/hooks/Debounce/Debounce.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * @see {@link https://dev.to/gabe_ragland/debouncing-with-react-hooks-jci}
+ */
+export const useDebounce = <T>(value: T, delay: number) => {
```

Comment:
Reckon we should just use something like https://github.com/xnimorz/use-debounce?